x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/parser/StatementParserTests.java

```diff
@@ -3138,6 +3188,32 @@ public void testInvalidJoinPatterns() {
             );
         }
 
+        // If one or more patterns participating in LOOKUP JOINs are partially quoted, we expect the partial quoting
+        // error messages to take precedence over any LOOKUP JOIN error messages.
+        {
+            // Generate a syntactically invalid (partial quoted) pattern.
+            var fromPatterns = randomIdentifier() + ":" + quote(randomIndexPatterns(without(CROSS_CLUSTER)));
```

Comment:
nit: for even more completeness, I'd also add the partially quoted case where the cluster name is quoted but the index pattern is unquoted. Same below, and also applies to the partial quoting in case of selectors.